### PR TITLE
Add PR checklist into PR template to update docs references in clients

### PR DESCRIPTION
**Pull request by sleshchenko**
> Read our [Contribution guide](https://github.com/eclipse/che-docs/blob/master/CONTRIBUTING.adoc) before submitting a PR.

### What does this PR do?
Add PR checklist into PR template to update docs references in clients

### What issues does this PR fix or reference?
It does not fix but is related to https://github.com/eclipse/che/issues/17739

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -6,4 +6,13 @@
...
+### Specify the version of the product this PR applies to.
+
+
+### PR Checklist
+
+As the author of this Pull Request I made sure that:
+
+- [ ] Changed articles references are updated where they are used:
+    - [ ] Dashboard [branding.constant.ts](https://github.com/eclipse/che-dashboard/blob/master/src/components/branding/branding.constant.ts) + [product.json](https://github.com/eclipse/che-dashboard/blob/master/src/assets/branding/product.json)
+    - [ ] Chectl
```
Comment (sleshchenko) [inline .github/PULL_REQUEST_TEMPLATE.md]: @tolusha Does chectl have one/a couple files which collects all documentation link chectl references to?

Comment (tolusha) [reply .github/PULL_REQUEST_TEMPLATE.md]: https://github.com/che-incubator/chectl/blob/master/src/constants.ts

Comment (sleshchenko) [reply .github/PULL_REQUEST_TEMPLATE.md]: thx. I've added that into checklist

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -6,4 +6,13 @@
...
 
-### Specify the version of the product this PR applies to. 
+### Specify the version of the product this PR applies to.
+
+
+### PR Checklist
+
+As the author of this Pull Request I made sure that:
+
+- [ ] Changed articles references are updated where they are used:
```
Comment (rkratky) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
- [ ] Changed article references are updated where they are used (or a redirect has been set up on the docs side):
```

**Review by rkratky — APPROVED** [review]
@sleshchenko, thanks. I just added one note as a suggestion.

**Pull request merged**